plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1482,6 +1503,42 @@ public static function get_local_woo_themes() {
 		return $woo_themes;
 	}
 
+	/**
+	 * Get the connected user's subscriptions.
+	 *
+	 * @return array
+	 */
+	public static function get_checked_products() {
```

Comment:
@gedex I've noticed that function is getting called even when the site is not connected. Is this required for non-connected sites? 

We can save a considerable amount of API calls if we limit this call to connected sites.

Reply:
> @gedex I've noticed that function is getting called even when the site is not connected. Is this required for non-connected sites?

Good point.

We need the site to be connected as the subscription data is needed. I'll add a check on the caller.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/47697/commits/76cc0415010edf43b50b94897ca4fc9bf34543d6.